doc/source/tune.rst

```diff
@@ -0,0 +1,160 @@
+Ray.tune: Efficient distributed hyperparameter search with Ray
+==============================================================
+
+This document describes Ray.tune, a hyperparameter tuning tool for long-running tasks such as RL and deep learning training. It has the following features:
+
+-  Early stopping algorithms such as `Median Stopping Rule <https://research.google.com/pubs/pub46180.html>`__ and `HyperBand <https://arxiv.org/abs/1603.06560>`__.
+
+-  Integration with visualization tools such as `TensorBoard <https://www.tensorflow.org/get_started/summaries_and_tensorboard>`__, `rllab's VisKit <https://media.readthedocs.org/pdf/rllab/latest/rllab.pdf>`__, and a `parallel coordinates visualization <https://en.wikipedia.org/wiki/Parallel_coordinates>`__.
+
+-  Flexible trial variant generation, including grid search, random search, and conditional parameter distributions.
+
+-  Resource-aware scheduling, including support for concurrent runs of algorithms that may themselves be parallel and distributed.
+
+
+Getting Started
+--------------- 
+
+::
+
+    from ray.tune import register_trainable, grid_search, run_experiments
+
+    def my_func(config, reporter):
+        import time, numpy as np
+        i = 0
+        while True:
+            reporter(timesteps_total=i, mean_accuracy=i ** config["alpha"])
+            i += config["beta"]
+            time.sleep(.01)
+
+    register_trainable("my_func", my_func)
+
+    run_experiments({
+        "my_experiment": {
+            "run": "my_func",
+            "resources": { "cpu": 1, "gpu": 0 },
+            "stop": { "mean_accuracy": 100 },
+            "config": {
+                "alpha": grid_search([0.2, 0.4, 0.6]),
+                "beta": grid_search([1, 2]),
+            },
+        }
+    })
+
+
+This script runs a small grid search over the ``my_func`` function using ray.tune, reporting status on the command line until the stopping condition of ``mean_accuracy >= 100`` is reached:
```

Comment:
We should be (if not already) explicit that this only handles increasing quantities.

Reply:
Done